File: app/src/main/java/com/shveed/cookmegood/recipe_page_fragments/IngredientFragment.java
package com.shveed.cookmegood.recipe_page_fragments;


import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.shveed.cookmegood.IngredientsGridAdapter;
import com.shveed.cookmegood.entity.Ingredient;
import com.shveed.wallpapperparser.R;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class IngredientFragment extends Fragment {

    List<Ingredient> ingredients;

    private String[] data = {"Помидоры", "Салат", "Хлеб", "Майонез", "Чеснок", "Сыр", "Укроп", "Лук"};

    private String[] amount = {"400г", "200г", "1 буханка", "200г", "2 головки", "300г", "50г", "50г"};

    private HashMap<Ingredient, Integer> buyMap = new HashMap<>();

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        ingredients = new ArrayList<>();
        for(int i = 0; i < 8; i++){
            Ingredient ingredient = new Ingredient(data[i], amount[i]);
            this.ingredients.add(ingredient);
        }
        View view = inflater.inflate(R.layout.f_ingredient, container, false);
        RecyclerView recyclerView = (RecyclerView) view.findViewById(R.id.gridIngred);
        IngredientsGridAdapter adapter = new IngredientsGridAdapter(getActivity(), ingredients);
        recyclerView.setLayoutManager(new GridLayoutManager(getActivity(), 2));
        recyclerView.setAdapter(adapter);
        return view;
    }

    public HashMap<Ingredient, Integer> getBuyMap() {
        return buyMap;
    }
}

File: app/src/main/java/com/shveed/cookmegood/fragment/IngredientFragment.kt
package com.shveed.cookmegood.fragment
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.shveed.cookmegood.adapter.IngredientsGridAdapter
import com.shveed.cookmegood.data.dto.Ingredient
import com.shveed.wallpapperparser.R
import java.util.*


class IngredientFragment : Fragment() {
    private var ingredients = arrayListOf<Ingredient>()
    private val data = arrayOf("Помидоры", "Салат", "Хлеб", "Майонез", "Чеснок", "Сыр", "Укроп", "Лук")
    private val amount = arrayOf("400г", "200г", "1 буханка", "200г", "2 головки", "300г", "50г", "50г")
    private val buyMap = HashMap<Ingredient, Int>()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        for (i in 0..7) {
            val ingredient = Ingredient(data[i], amount[i])
            ingredients.add(ingredient)
        }
        val view = inflater.inflate(R.layout.f_ingredient, container, false)
        val recyclerView = view.findViewById<View>(R.id.gridIngred) as RecyclerView
        val adapter = IngredientsGridAdapter(activity, ingredients)
        recyclerView.layoutManager = GridLayoutManager(activity, 2)
        recyclerView.adapter = adapter
        return view
    }
    fun getBuyMap(): HashMap<Ingredient, Int>? {
        return buyMap
    }
}